processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -397,6 +408,34 @@ private void reportCannotCreateMapping() {
             }
         }
 
+        /**
+         * Reports that a mapping could not be created.
+         *
+         * @param method the method that should be mapped
+         * @param posHint hint which @Mapping is the culprit
+         * @param sourceErrorMessagePart the error message part for the source
+         * @param sourceType the source type of the mapping
+         * @param targetType the type of the target mapping
+         * @param targetPropertyName the name of the target property
+         */
+        void reportCannotCreateCollectionOrMapUpdateMapping(Method method, AnnotationMirror posHint,
+                                                            String sourceErrorMessagePart,
+                                                            Type sourceType, Type targetType,
+                                                            String targetPropertyName ) {
+            ctx.getMessager().printMessage(
+                method.getExecutable(),
+                posHint,
+                targetWriteAccessorType == TargetWriteAccessorType.ADDER ?
+                    Message.PROPERTYMAPPING_CANNOT_CREATE_UPDATE_ADDER
+                    : Message.PROPERTYMAPPING_CANNOT_CREATE_UPDATE_MAP_OR_COLLECTION,
```

Comment:
@filiphr .. I'm confused.. It seems that if you have an update mapping, MapStruct forges a regular collection mapping to call. However, if you selected "ADDER"  it will not do so and give the vague error message.

It feels inconsistent. I think what we should do is  **always** forge a regular-collection-(non-update)-mapping method also in the case of adder.  Reasoning.. we can't revert this behaviour. This means this PR will possibly end up in the examples how to do update methods.. So.. than we can also close #487 and #489 (the unit test shows how tricky this is, and I only covered one case)..

Another thought is to include a util library.. I've some ideas here.. Perhaps we should go forward.. Shall I make a demonstrator?